src/hoc/useNodes.tsx

```diff
@@ -0,0 +1,16 @@
+import { useSelector as reduxUseSelector, useDispatch, TypedUseSelectorHook } from 'react-redux';
+import { addNode, AddNode } from '@/module/node/actions';
+import { IGraphState, INodesState } from '@/type/types';
+
+const useSelector: TypedUseSelectorHook<IGraphState> = reduxUseSelector;
+
+export default () => {
+  // Replaces mapStateToProps
+  const nodes: INodesState = useSelector((state) => state.nodes);
```

Comment:
Just pulled the branch to play with it. 
Do we even need hoc's anymore? Can we just export the individual hooks and import it to the file we need them in?